src/main/java/com/khutircraftubackend/jwt/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,56 @@
+package com.khutircraftubackend.jwt;
+
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.khutircraftubackend.config.UserDetailsConfig;
+import com.khutircraftubackend.security.UserDetailsImpl;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private final JWTVerifier jwtVerifier;
+    private final UserDetailsConfig userDetailsConfig;
+
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request,
+                                    @NonNull HttpServletResponse response,
+                                    @NonNull FilterChain filterChain)
+            throws ServletException, IOException {
+
+        String authHeader = request.getHeader("Authorization");
+
+        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+            String token = authHeader.substring(7);
+            DecodedJWT jwt = jwtVerifier.verify(token);
+            String email = jwt.getSubject();
+        if (email != null &&
+                SecurityContextHolder.getContext().getAuthentication() == null) {
```

Comment:
this implementation doesn't match the requirements in jira ticket,
could you kindly review it again, please?

Reply:
Thank you. Fixed it.